x-pack/plugins/security_solution/public/timelines/components/timeline/body/events/index.tsx

```diff
@@ -93,27 +84,25 @@ const EventsComponent: React.FC<Props> = ({
         browserFields={browserFields}
         columnHeaders={columnHeaders}
         columnRenderers={columnRenderers}
-        containerElementRef={containerElementRef}
-        docValueFields={docValueFields}
         event={event}
         eventIdToNoteIds={eventIdToNoteIds}
         getNotesByIds={getNotesByIds}
         isEventPinned={eventIsPinned({ eventId: event._id, pinnedEventIds })}
         isEventViewer={isEventViewer}
+        isExpanded={expanded.eventId === event._id}
         key={`${event._id}_${event._index}`}
         loadingEventIds={loadingEventIds}
         onColumnResized={onColumnResized}
         onPinEvent={onPinEvent}
         onRowSelected={onRowSelected}
         onUnPinEvent={onUnPinEvent}
-        onUpdateColumns={onUpdateColumns}
         refetch={refetch}
         rowRenderers={rowRenderers}
+        onEventToggled={() => onEventToggled(event)}
```

Comment:
To avoid this lambda function you can do that in `x-pack/plugins/security_solution/public/timelines/components/timeline/body/events/stateful_event.tsx`

1. Change props of `onEventToggled` to `(event: TimelineItem) => void;`
2. add this callback
```
const handleOnEventToggled = useCallback(() => onEventToggled(event), [event, onEventToggled]);
```
3. Now, you can replace `onEventToggled={() => onEventToggled(event)}` to `onEventToggled={onEventToggled}`

Reply:
what about we pass both event and onEventToggled all the way down to `x-pack/plugins/security_solution/public/timelines/components/timeline/body/actions/index.tsx`.

so onEventToggled can access `event` naturally with no params required, and then we could wrap onEventToggled with useCallback and pass it to `EuiButtonIcon`